### PR TITLE
Fix the plugin

### PR DESCRIPTION
**Pull request by Noarkhh**

### CODE REVIEW

File: lib/membrane_ivf/serializer.ex [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+              ],
+              height: [
+                spec: non_neg_integer() | nil,
+                default: nil,
+                description: """
+                Height of a frame, needed if not provided with stream format. If it's not specified either in this option or the stream format, the element will crash.
+                """
+              ],
+              timebase: [
+                spec: {non_neg_integer(), pos_integer()},
```
Comment (varsill) [inline lib/membrane_ivf/serializer.ex:30]: can it really be {0, <some positive number>}?

**Review by varsill — CHANGES_REQUESTED** [review]

File: test/serializer_test.exs [hunk 2]
```diff
@@ -66,15 +65,17 @@ defmodule Membrane.Element.IVF.SerializerTest do
...
-      ivf_buffer.payload
+    <<file_header::binary-size(32)>> = ivf_file_header_buffer.payload
+
+    <<frame_header::binary-size(12), _frame::binary>> = ivf_buffer.payload
 
     <<signature::binary-size(4), version::binary-size(2), length_of_header::binary-size(2),
       four_cc::binary-size(4), width::binary-size(2), height::binary-size(2),
       time_base_denominator::binary-size(4), time_base_numerator::binary-size(4),
-      number_of_frames::binary-size(4), _unused::binary-size(4)>> = file_header
+      _number_of_frames::binary-size(4), _unused::binary-size(4)>> = file_header
```
[the remark this answers is not in the document]
Comment (Noarkhh) [reply test/serializer_test.exs:78]: now the element actually counts the number of frames processed and on eos seeks and updates this field. We assert on it later (lines 104-106)

File: test/integration_test.exs [hunk 3]
```diff
@@ -1,32 +1,28 @@
...
-  @input_video_vp9 %{path: "./test/fixtures/input_vp9.ivf", width: 1080, height: 720}
-  @results_dir "./test/results/"
-  @result_file_vp8 "result_vp8.ivf"
-  @result_file_vp9 "result_vp9.ivf"
+  @input_path_vp8 "./test/fixtures/input_vp8.ivf"
+  @input_path_vp9 "./test/fixtures/input_vp9.ivf"
+  @output_file_vp8 "output_vp8.ivf"
+  @output_file_vp9 "output_vp9.ivf"
 
   defmodule TestPipeline do
```
[the remark this answers is not in the document]
Comment (Noarkhh) [reply test/integration_test.exs:14]: no, forgor to remove

File: test/serializer_test.exs [hunk 4]
```diff
@@ -114,7 +112,7 @@ defmodule Membrane.IVF.SerializerTest do
 
   @tag :tmp_dir
   test "serialize real vp9 buffers", %{tmp_dir: tmp_dir} do
-    buffers = File.read!(Path.join(@fixtures_dir, "capture.dump")) |> :erlang.binary_to_term()
+    buffers = Path.join(@fixtures_dir, "capture.dump") |> File.read!() |> :erlang.binary_to_term()
```
Comment (varsill) [inline test/serializer_test.exs:115]: I cannot comment on line `132` of that file, but there are yet another some places where such a change with piping would be welcome ;)

**Review by varsill — APPROVED** [review]
🥇

**Pull request merged**